File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        reverseHelper(s, 0, s.size() - 1);
    }

private:
    void reverseHelper(vector<char>& s, int left, int right) {
        if (left < right) {
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            reverseHelper(s, left + 1, right - 1);
        }
    }
};

File: 0344-reverse-string/0344-reverse-string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        self.reverseHelper(s, 0, len(s) - 1)

    def reverseHelper(self, s: List[str], left: int, right: int) -> None:
        if left < right:
            s[left], s[right] = s[right], s[left]
            self.reverseHelper(s, left + 1, right - 1)